test: get kvstore and bugtool output

ianvernon opened the pull request:
* Get output of `cilium kvstore get cilium --recursive` in both runtime and
Kubernetes test suites.
* Get debugtool output and bugtool archive for each Cilium pod in the
Kubernetes test suite.

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #2746 
Fixes: #3006

Commit:
test: get kvstore, debuginfo, and bugtool output

* Get output of `cilium kvstore get cilium --recursive` in both runtime and
Kubernetes test suites.
* Get bugtool archive for each Cilium pod in the Kubernetes test suite.
* Remove invocation of `cilium debuginfo` from the log-gaterhing code in Ginkgo
because this is already covered by `cilium-bugtool`.

Signed-off by: Ian Vernon <ian@cilium.io>

eloycoto commented on test/helpers/cons.go:
```diff
@@ -148,28 +148,31 @@ const (
-	"cilium policy get":               "policy_get.txt",
-	"cilium status --all-controllers": "status.txt",
-	"sudo cilium debuginfo":           "debuginfo.txt",
+	"cilium endpoint list -o json":          "endpoint_list.txt",
```
Hi, 

I do not liked the JSON output here. We should be able to debug endpoints, services as a user will do. The output of `-json` is a bit different, and we should be able to know problems without getting the JSON output. I'm happy if you add a new line without JSON output.

ianvernon replied:
I didn't change this, this was just due to a reformatting with `go fmt`. The output of `cilium endpoint list` is contained in the output of `debuginfo.txt`, which should be one of the primary sources for data so we can dogfood it.

eloycoto replied:
ok!

Commit:
test/helpers: fix improper invocation of cilium-bugtool

Running `cilium bugtool` is not valid, while `cilium-bugtool` is.

Signed-off by: Ian Vernon <ian@cilium.io>

eloycoto commented on test/helpers/kubectl.go:
```diff
@@ -697,33 +692,22 @@ func (kub *Kubectl) CiliumReportDump(namespace string) {
+		if res.WasSuccessful() {
+			// Default output directory is /tmp for bugtool.
+			res = kub.Exec(fmt.Sprintf("kubectl exec -n %s %s -- ls /tmp/", namespace, pod))
+			tmpList := strings.Split(strings.TrimSuffix(res.stdout.String(), "\n"), "\n")
```
Here you can use `res.ByLines` no?

ianvernon replied:
Done.

tgraf commented on test/helpers/cilium.go:
```diff
@@ -551,7 +551,7 @@ func (s *SSHMeta) DumpCiliumCommandOutput() {
 
 	// No need to create file for bugtool because it creates an archive of files
 	// for us.
-	res := s.ExecCilium(fmt.Sprintf("bugtool -t %s", filepath.Join(BasePath, testPath)))
```
Sorry, I didn't see this. I grep'ed for `cilium bugtool` and didn't find anything.
